src/lib/transaction_snark/test/zkapps_examples/thing/thing.ml

```diff
@@ -0,0 +1,183 @@
+let sk = Private_key.create ()
+
+let pk = Public_key.of_private_key_exn sk
+
+let pk_compressed = Public_key.compress pk
+
+(* create account id from token + pubkey *)
+
+let account_id = Account_id.create pk_compressed Token_id.default
+
+(* we want to create a circuit with a domain of size 2^16 *)
+let num_constraints = 1 lsl 16
+
+let () = Format.printf "num_constraints: %d@." num_constraints
+
+(* create a zkapp? but the rules are wrap circuits, mmm... *)
+
+let tag, _cache, _p_module, Pickles.Provers.[ prover ] =
+  Zkapps_examples.compile () ~cache:Cache_dir.cache ~auxiliary_typ:Impl.Typ.unit
+    ~branches:(module Nat.N1)
+    ~max_proofs_verified:(module Nat.N0)
+    ~name:"empty_update"
+    ~constraint_constants:
+      (Genesis_constants.Constraint_constants.to_snark_keys_header
+         constraint_constants )
+    ~choices:(fun ~self:_ ->
+      [ Zkapps_big_circuit.rule ~num_constraints pk_compressed ] )
+
+(* get verification key from memory *)
+
+let vk : Side_loaded_verification_key.t =
+  Pickles.Side_loaded.Verification_key.of_compiled tag
+
+(* I couldn't figure out how to get the domain size from vk here
+   let () =
+      let domain_size = vk.step.domain_size in
+      assert (domain_size = 16)
+*)
+
+(* create a block_on_async_exn prover? by just wrappring the prover? but it says it creates an account update... i'm confused... *)
+
+let account_update, () = Async.Thread_safe.block_on_async_exn prover
+
+(* create a deploy account update with the zkapp in it *)
+
+let deploy_account_update_body : Account_update.Body.t =
+  (* TODO: This is a pain. *)
+  { Account_update.Body.dummy with
+    public_key = pk_compressed
+  ; update =
+      { Account_update.Update.dummy with
+        verification_key =
+          Set
+            { data = vk
+            ; hash =
+                (* TODO: This function should live in
+                   [Side_loaded_verification_key].
+                *)
+                Zkapp_account.digest_vk vk
+            }
+      }
+  ; preconditions =
+      { Account_update.Preconditions.network =
+          Zkapp_precondition.Protocol_state.accept
+      ; account = Accept
+      }
+  ; caller = Token_id.default
+  ; use_full_commitment = true
+  ; authorization_kind = Signature
+  }
+
+let deploy_account_update : Account_update.t =
+  (* TODO: This is a pain. *)
+  { body = deploy_account_update_body
+  ; authorization = Signature Signature.dummy
+  }
+
+(* why are there two updates in there? an [account_update] and then a [deploy_account_update] *)
+
+let account_updates =
+  []
+  (* this looks like a call to prover *)
+  |> Zkapp_command.Call_forest.cons_tree account_update
+  (* and this like a normal account update call *)
+  |> Zkapp_command.Call_forest.cons deploy_account_update
+
+let memo = Signed_command_memo.empty
+
+(* what's a transaction commitment? (looks like it's just a field element which is a hash (poseidon) of the call forest (which is just `hash([prev_hash, new_el])`)) *)
```

Comment:
It's a hash of the transaction's contents, which the parties then sign to authorize.